lib/charms/opensearch/v0/opensearch_tls.py

```diff
@@ -130,19 +130,26 @@ def _on_certificate_available(self, event: CertificateAvailableEvent) -> None:
         old_cert = secrets.get("cert", None)
         renewal = old_cert is not None and old_cert != event.certificate
 
+        if not isinstance(event.chain, list):
+            # We should have got a list at this point
+            logging.debug("[WARN] Expected CA chain to be a list, received a string instead")
+            ca_chain = event.chain
+        else:
+            ca_chain = "\n".join(event.chain[::-1])
```

Comment:
Let's convert the list to a str from the get go, and remove the checks on type

Reply:
That is exactly what is happening right now. And the checks are never meant to be executed.

Now my hesitation comes from the fact that some of the related calls can be executed also when we already have secrets stored. And I feared that the call may be possible to be invoked both with a fresh (TLS charm delivered) certificate, and with a cert that comes from our own, already processed secret.

If it's 1000% that this could NEVER happen, we can remove the checks of course.

Reply:
If you check the source of `CertificateAvailableEvent` - you'll notice `event.chain` is of type `List`. It is NOT a string.
It is safe to trust the input of type List.